src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -560,36 +560,40 @@ private EnvironmentOptions GetEnvironmentOptions(bool requiresNetFramework, stri
             var os = DetermineOSPlatform();
             EnvironmentOptions options = new EnvironmentOptions();
 
-            if (os == OSPlatform.Linux || os == OSPlatform.OSX)
-            {
-                if (requiresNetFramework)
-                {
-                    options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, Constants.MsBuildCommandName);
-                }
-            }
-
             //We want to provide the MsBuild path only if it's a framework solution. Buildalyzer automatically builds core solutions using "dotnet"
             if (requiresNetFramework)
             {
-                try
+                if (os == OSPlatform.Linux || os == OSPlatform.OSX)
                 {
-                    var msbuildExe = _analyzerConfiguration.BuildSettings.MSBuildPath;
-                    if (string.IsNullOrEmpty(msbuildExe))
+                    options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, Constants.MsBuildCommandName);
+                }
+                else {
+                    try
                     {
-                        msbuildExe = _msBuildDetector.GetFirstMatchingMsBuildFromPath(toolsVersion: toolsVersion);
+                        var msbuildExe = _analyzerConfiguration.BuildSettings.MSBuildPath;
+                        if (string.IsNullOrEmpty(msbuildExe))
+                        {
+                            msbuildExe = _msBuildDetector.GetFirstMatchingMsBuildFromPath(toolsVersion: toolsVersion);
+                        }
+                        if (!string.IsNullOrEmpty(msbuildExe)) 
+                        {
+                            options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, msbuildExe);
+                        }
+                        else 
+                        { 
+                            throw new NullReferenceException(); 
+                        }
                     }
-                    if (!string.IsNullOrEmpty(msbuildExe)) options.EnvironmentVariables.Add(EnvironmentVariables.MSBUILD_EXE_PATH, msbuildExe);
-                    else { throw new Exception(); }
-
-                }
-                catch (Exception ex)
-                {
-                    Logger.LogError(ex, "Build error: Codelyzer wasn't able to retrieve the MSBuild path");
+                    catch (NullReferenceException ex)
+                    {
+                        Logger.LogError(ex, "Build error: Codelyzer wasn't able to retrieve the MSBuild path. " +
+                            "Visual Studio and MSBuild might not be installed.");
```

Comment:
We should be able to just log the exception object here if the error message is added directly to the exception. Also, remove the "Build error: " prefix to prevent the error message from being categorized incorrectly.